tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,198 @@
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have the expected entries in vmi status", func() {
+			expectedInterfaces := []kvirtv1.VirtualMachineInstanceNetworkInterface{
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        primaryInterfaceMac,
+					Name:       primaryNetwork,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceDomainAndGA,
+					InterfaceName: "eth1",
+					MAC:           secondaryInterface1Mac,
+					Name:          secondaryInterface1Name,
+				},
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        secondaryInterface2Mac,
+					Name:       secondaryInterface2Name,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "eth0",
+					MAC:           primaryInterfaceNewMac,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: dummyInterfaceName,
+					MAC:           dummyInterfaceMac,
+				},
+			}
+
+			var vmiObj *kvirtv1.VirtualMachineInstance
+			Eventually(func() bool {
+				var err error
+				vmiObj, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				return dummyInterfaceExists(vmiObj)
+			}, 120*time.Second, 2*time.Second).Should(Equal(true))
+
+			Expect(len(vmiObj.Status.Interfaces)).To(Equal(len(expectedInterfaces)),
+				fmt.Sprintf("interfaces doesn't match\n%v\n%v", expectedInterfaces, vmiObj.Status.Interfaces))
+
+			orderedInterfaces := orderInterfaceStatus(vmiObj.Status.Interfaces)
+			for i := range orderedInterfaces {
+				expectedInterfaces[i].IP = orderedInterfaces[i].IP
+				expectedInterfaces[i].IPs = orderedInterfaces[i].IPs
+			}
+
+			Expect(reflect.DeepEqual(expectedInterfaces, orderedInterfaces)).To(BeTrue(),
+				fmt.Sprintf("interfaces doesn't match\n%v\n%v", expectedInterfaces, orderedInterfaces))
+		})
+	})
+})
+
+func newNetworkAttachmentDefinition(networkName string) *nadv1.NetworkAttachmentDefinition {
+	config := fmt.Sprintf(`{"cniVersion": "0.3.1", "name": "%s", "type": "cnv-bridge", "bridge": "%s"}`, networkName, networkName)
+	return &nadv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: networkName,
+		},
+		Spec: nadv1.NetworkAttachmentDefinitionSpec{Config: config},
+	}
+}
+
+// dummyInterfaceExists waits for least the dummy interface to be reported by guest agent,
+// which means there was a guest-agent report finally, and then we can compare the rest expected info.
+func dummyInterfaceExists(vmi *kvirtv1.VirtualMachineInstance) bool {
+	for i := range vmi.Status.Interfaces {
+		if vmi.Status.Interfaces[i].InterfaceName == dummyInterfaceName {
+			return true
+		}
+	}
+	return false
+}
+
+// orderInterfaceStatus orders "guest-agent" only records in predictable order
+func orderInterfaceStatus(interfaces []kvirtv1.VirtualMachineInstanceNetworkInterface) []kvirtv1.VirtualMachineInstanceNetworkInterface {
+	orderedArray := interfaces
```

Comment:
Both point to the same list, so I do not see the point.

Reply:
arrays are copied by value in golang
https://go.dev/play/p/oZN8fCbuUVn
so now the function returns a new instance

Reply:
right, confused with arrays that are copied by values, but since it is a slice it isn't